go/pkg/client/tree.go

```diff
@@ -89,6 +164,20 @@ func loadFiles(execRoot string, excl []*command.InputExclusion, path string, fs
 			},
 		}
 		return nil
+	} else if t == command.SymlinkInputType {
+		fs[path] = &fileSysNode{
+			symlink: &symlinkNode{target: meta.Symlink.Target},
```

Comment:
I think there is a problem here -- if the symlink target is absolute (and we have Preserve=true), then even when it is actually under the exec root we cannot preserve it as is, because that would fail on the bot (because the bot will map the exec root to a different absolute path).
In this edge case, I think that we will need to convert the symlink target to a relative path instead.

Alternatively, we could disallow absolute symlink targets in general when Preserve=true.

Reply:
Also, I suspect that when you fix this, you will no longer need the exec root placeholder workaround in tests.
But if you do want to test absolute symlinks, I think the IsAbsolute attribute of inputPath already does what you want, you don't need the exec root placeholder.

Reply:
Thanks for pointing this out. Yeah the test is simplified..

> I think the IsAbsolute attribute of inputPath already does what you want, you don't need the exec root placeholder.

Note that `IsAbsolute` only affects how the test data  are constructed in the filesystem, but doesn't apply to the `SymlinkNode` protos. Anyway, the placeholder is removed now.